clippy_lints/src/types.rs

```diff
@@ -1157,6 +1157,28 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for CastPass {
             lint_fn_to_numeric_cast(cx, expr, ex, cast_from, cast_to);
             if let ExprKind::Lit(ref lit) = ex.node {
                 use syntax::ast::{LitIntType, LitKind};
+                if let LitKind::Int(n, _) = lit.node {
+                    if cast_to.is_fp() {
+                        let from_nbits = int_ty_to_nbits(cast_from, cx.tcx);
+                        let to_nbits = if let ty::Float(FloatTy::F32) = cast_to.sty {
+                            32
+                        } else {
```

Comment:
I would prefer a `match` here, just to make sure, that this will work in the future if there are more `FloatTy` added. Note that you also have to `match` on `Infer(Float(_)) => 64 (?)`.

Reply:
done